--- src/Makefile.in.orig	2022-04-04 15:19:35.000000000 -0700
+++ src/Makefile.in	2022-04-04 15:20:04.000000000 -0700
@@ -347,7 +347,7 @@
 
 ## ARM Macs require that all code have a valid signature.  Since pdump
 ## invalidates the signature, we must re-sign to fix it.
-DO_CODESIGN=$(patsubst aarch64-apple-darwin%,yes,@configuration@)
+DO_CODESIGN=$(patsubst arm64-apple-darwin%,yes,@configuration@)
 
 -include ${top_builddir}/src/verbose.mk
 
