--- nextstep/Makefile.in.orig	2022-04-04 15:33:50.000000000 -0700
+++ nextstep/Makefile.in	2022-04-04 15:35:13.000000000 -0700
@@ -66,9 +66,8 @@
 
 # FIXME: Don't install the dump file into the app bundle when
 # self-contained install is disabled.
-${ns_applibexecdir}/Emacs.pdmp: ${ns_appdir} ${ns_check_file} ../src/emacs${EXEEXT}.pdmp
+${ns_applibexecdir}/Emacs.pdmp: ${ns_appdir} ${ns_check_file}
 	${MKDIR_P} ${ns_applibexecdir}
-	cp -f ../src/emacs${EXEEXT}.pdmp $@
 
 .PHONY: FORCE
 
