diff --git a/Makefile.in b/Makefile.in
index a89836dca2c..9e4838a72e7 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -643,7 +631,9 @@ install-arch-dep:
 	cd "$(DESTDIR)${bindir}" && $(LN_S_FILEONLY) "$(EMACSFULL)" "$(EMACS)"
 endif
 else
+ifeq (${DUMPING},pdumper)
 	${INSTALL_DATA} src/emacs.pdmp "$(DESTDIR)${libexecdir}/Emacs.pdmp"
+endif
 	subdir=${ns_appresdir}/site-lisp && ${write_subdir}
 	rm -rf ${ns_appresdir}/share
 endif
