diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS emacs-28.1.orig/lisp/startup.el emacs-28.1/lisp/startup.el
--- emacs-28.1.orig/lisp/startup.el	2022-03-11 08:04:21.000000000 +0100
+++ emacs-28.1/lisp/startup.el	2022-07-19 16:30:18.494908537 +0200
@@ -528,6 +528,8 @@
 It sets `command-line-processed', processes the command-line,
 reads the initialization files, etc.
 It is the default value of the variable `top-level'."
+  (setq configure-info-directory (string-replace "\0" "" configure-info-directory))
+  (setf (car Info-default-directory-list) configure-info-directory)
   (if command-line-processed
       (message internal--top-level-message)
     (setq command-line-processed t)
