diff --git a/m4/fchmodat.m4 b/m4/fchmodat.m4
index e4688708063..611bb23f09e 100644
--- a/m4/fchmodat.m4
+++ b/m4/fchmodat.m4
@@ -64,9 +64,9 @@ AC_DEFUN
                 result |= 4;
               return result;
             ]])],
-         [gl_cv_func_fchmodat_works=yes],
+         [gl_cv_func_fchmodat_works=no],
          [case $? in
-            2) gl_cv_func_fchmodat_works='nearly' ;;
+            2) gl_cv_func_fchmodat_works=no ;;
             *) gl_cv_func_fchmodat_works=no ;;
           esac
          ],

henry@holly:~/build/conda/emacs-feedstock/recipe$ 
henry@holly:~/build/emacs$ git diff
diff --git a/m4/fchmodat.m4 b/m4/fchmodat.m4
index e4688708063..611bb23f09e 100644
--- a/m4/fchmodat.m4
+++ b/m4/fchmodat.m4
@@ -64,9 +64,9 @@ AC_DEFUN
                 result |= 4;
               return result;
             ]])],
-         [gl_cv_func_fchmodat_works=yes],
+         [gl_cv_func_fchmodat_works=no],
          [case $? in
-            2) gl_cv_func_fchmodat_works='nearly' ;;
+            2) gl_cv_func_fchmodat_works=no ;;
             *) gl_cv_func_fchmodat_works=no ;;
           esac
          ],

