diff -aur a/configure.ac b/configure.ac
--- a/configure.ac	2025-01-15 23:56:23 +0000
+++ b/configure.ac	2025-01-15 23:55:47 +0000
@@ -4593,11 +4593,11 @@
        WITH_IFAVAILABLE="$WITH_IFAVAILABLE --with-tiff=ifavailable";;
   esac
 fi
-case $with_gnutls,$HAVE_GNUTLS in
-  no,* | ifavailable,* | *,yes) ;;
-  *) MISSING="$MISSING gnutls"
-     WITH_IFAVAILABLE="$WITH_IFAVAILABLE --with-gnutls=ifavailable";;
-esac
+#case $with_gnutls,$HAVE_GNUTLS in
+#  no,* | ifavailable,* | *,yes) ;;
+#  *) MISSING="$MISSING gnutls"
+#     WITH_IFAVAILABLE="$WITH_IFAVAILABLE --with-gnutls=ifavailable";;
+#esac
 case $with_json,$HAVE_JSON in
   no,* | ifavailable,* | *,yes) ;;
   *) MISSING="$MISSING json"
